### PR TITLE
fix: writing non-existing file

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Fixed issue with writing not existing file.

Added logging and fixed issue with copilot plugin not available during new project loading.

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt [hunk 1]
```diff
@@ -1,49 +1,82 @@
...
     private val vfsFile: VirtualFile?
 
     init {
-        val file = File(data["file"] as String)
-        if (!isFileInsideProject(project, file)) {
-            throw Exception("File is not a part of a project")
+        if (isFileInsideProject(project, ioFile)) {
+            vfsFile = VfsUtil.findFileByIoFile(ioFile, true)
+        } else {
+            LOG.warn("File $ioFile is not a part of a project")
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt:29]: Does this log a warning every time for a new file?

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt [hunk 2]
```diff
@@ -1,49 +1,82 @@
...
     private val undoLabel: String? = data["undoLabel"] as String?
-    private val vfsDoc: Document?
+    private val ioFile: File = File(data["file"] as String)
     private val vfsFile: VirtualFile?
 
     init {
-        val file = File(data["file"] as String)
-        if (!isFileInsideProject(project, file)) {
-            throw Exception("File is not a part of a project")
+        if (isFileInsideProject(project, ioFile)) {
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt:26]: If the `getRealPath` method is copied from copilot, you might want to copy a new version which supports adding a file also to a new directory..

Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt:26]: Updated, and added support for parent directories creation

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt (hunk 1 above)
Comment (MarcinVaadin) [reply src/main/kotlin/com/vaadin/plugin/copilot/handler/WriteFileHandler.kt:29]: No, only if new file path is not subpath of project base directory. Check `Help > Open Log in Editor` to see logs

File: src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt [hunk 3]
```diff
@@ -28,21 +29,8 @@ abstract class AbstractHandler(val project: Project) : Runnable {
...
-    @Throws(IOException::class)
-    private fun getRealPath(file: File): Path {
-        val path = file.toPath()
-        return try {
-            path.toRealPath()
-        } catch (e: NoSuchFileException) {
-            // As we allow creating new files, we check the directory instead
-            path.parent.toRealPath().resolve(path.fileName)
-        }
+        return project.basePath?.let { file.path.startsWith(it) } == true
```
Comment (Artur-) [inline src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt:33]: How does this make sure that file is not "/my/basepath/../../etc/passwd" ?

Comment (Artur-) [reply src/main/kotlin/com/vaadin/plugin/copilot/handler/AbstractHandler.kt:33]: The updated version in copilot is actually only in https://github.com/vaadin/copilot-internal/pull/882. It checks the parent's real path for the first parent it finds

**Review by Artur- — APPROVED** [review]

**Pull request merged**